inspirehep/modules/records/receivers.py

```diff
@@ -106,3 +114,24 @@ def match_valid_experiments(recid, json, *args, **kwargs):
                     # If the experiment cannot be matched it is considered valid
                     facet_experiment = exp.get("experiment")
                 exp.update({"facet_experiment": facet_experiment})
+
+
+@before_record_insert.connect
+@before_record_update.connect
+def dois_clean(sender, *args, **kwargs):
+    dois = []
+    new_l = []
+    if 'dois' in sender:
+        for element in sender['dois']:
+            if 'value' in element:
+                dois.append(element['value'])
+
+        if len(list(set(dois))) > 1:
+            for element in sender['dois']:
```

Comment:
I am bit confused with this. You have already iterated over `sender['dois']` before and added them to the `dois` list. Why do you need to iterate them again here and in the lines below?

If you just wanted to delete duplicates something like this would work?

```python
dois = list(set(dois))
```